docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == _from, "Wrong from address.");
+        require(_to != address(0), "Cannot send to 0x0.");
+        _transfer(_to, _tokenId);
+    }
+
+    function approve(address _approved, uint256 _tokenId) external override canOperate(_tokenId) validNFToken(_tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(_approved != tokenOwner);
+        idToApproval[_tokenId] = _approved;
+        emit Approval(tokenOwner, _approved, _tokenId);
+    }
+
+    function setApprovalForAll(address _operator, bool _approved) external override {
+        ownerToOperators[msg.sender][_operator] = _approved;
+        emit ApprovalForAll(msg.sender, _operator, _approved);
+    }
+
+    function balanceOf(address _owner) external view override returns (uint256) {
+        require(_owner != address(0));
+        return _getOwnerNFTCount(_owner);
+    }
+
+    function ownerOf(uint256 _tokenId) external view override returns (address _owner) {
+        require(idToOwner[_tokenId] != address(0));
+        _owner = idToOwner[_tokenId];
+    }
+
+    function getApproved(uint256 _tokenId) external view override validNFToken(_tokenId) returns (address) {
+        return idToApproval[_tokenId];
+    }
+
+    function isApprovedForAll(address _owner, address _operator) external override view returns (bool) {
+        return ownerToOperators[_owner][_operator];
+    }
+
+    function _transfer(address _to, uint256 _tokenId) internal {
+        address from = idToOwner[_tokenId];
+        _clearApproval(_tokenId);
+
+        _removeNFToken(from, _tokenId);
+        _addNFToken(_to, _tokenId);
+
+        emit Transfer(from, _to, _tokenId);
+    }
+
+    function randomIndex() internal returns (uint) {
+        uint totalSize = TOKEN_LIMIT - numTokens;
+        uint index = uint(keccak256(abi.encodePacked(nonce, msg.sender, block.difficulty, block.timestamp))) % totalSize;
+        uint value = 0;
+        if (indices[index] != 0) {
+            value = indices[index];
+        } else {
+            value = index;
+        }
+
+        // Move last value to selected position
+        if (indices[totalSize - 1] == 0) {
+            // Array position not initialized, so use position
+            indices[index] = totalSize - 1;
+        } else {
+            // Array position holds a value so use that
+            indices[index] = indices[totalSize - 1];
+        }
+        nonce++;
+        // Don't allow a zero index, start counting at 1
+        return value.add(1);
+    }
+
+    // Calculate the mint price
+    function getPrice() public view returns (uint) {
+        require(publicSale, "Sale not started.");
+        uint elapsed = block.timestamp.sub(saleStartTime);
+        if (elapsed >= saleDuration) {
+            return 0;
+        } else {
+            return saleDuration.sub(elapsed).mul(price).div(saleDuration);
+        }
+    }
```

Comment:
Only used for minting. Calculates the price based on a dutch auction initiated by startSale